components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/http/InboundHttpServerWorker.java

```diff
@@ -80,9 +83,27 @@ public InboundHttpServerWorker(int port, String tenantDomain, SourceRequest sour
         isInternalHttpsInboundEndpoint = (HTTPEndpointManager.getInstance().getInternalInboundHttpsPort() == port);
     }
 
+    InboundHttpServerWorker(int port, String tenantDomain, SourceRequest sourceRequest,
```

Comment:
Shall we create a seperate 'InboundCorrelationEnabledHttpServerWorker' extending 'InboundHttpServerWorker' and override only the run method to do the additional stuff and call super.run()?
This is to avoid having to chech whether correlation is enabled during each run

Reply:
done in https://github.com/wso2/micro-integrator/pull/1150/commits/11744a0818bc09c479218d379f83e776c6738ab5